zebra/interface.c

```diff
@@ -228,12 +228,16 @@ struct interface *if_lookup_by_index_per_ns(struct zebra_ns *ns,
 	struct route_node *rn;
 	struct interface *ifp = NULL;
 
-	if_build_key(ifindex, &p);
-	rn = route_node_lookup(ns->if_table, &p);
-	if (rn) {
-		ifp = (struct interface *)rn->info;
-		route_unlock_node(rn); /* lookup */
-	}
+	if (vrf_is_backend_netns()) {
+		if_build_key(ifindex, &p);
+		rn = route_node_lookup(ns->if_table, &p);
+		if (rn) {
+			ifp = (struct interface *)rn->info;
+			route_unlock_node(rn); /* lookup */
+		}
+	} else
+		ifp = if_lookup_by_index(
+				ifindex, VRF_UNKNOWN);
```

Comment:
nit: run clang-format here to put the everything in the same line.

Reply:
Actually I think we don't need to change this function. The interface lookup using `ns->if_table` should work for all VRF backends. What do you think?

Reply:
you are right.
I wonder if we should not catch people using wrong API ( for instance if_lookup_by_index should never be caled with vrf_id=VRF_UNKNOWN).